### PR TITLE
 Drop support for python2 

### LINKED ISSUE #138: Python2 support
I’d like to drop python2 support

### PR DESCRIPTION
**Pull request by ijl**
This closes #138. It depends on #404.

### CODE REVIEW

File: guide/src/class.md [hunk 1]
```diff
@@ -503,18 +503,14 @@ Each methods corresponds to python's `self.attr`, `self.attr = value` and `del s
...
 
-    On Python 3.x, provides the conversion to `bytes`.
-    On Python 2.7, `__bytes__` is allowed but has no effect.
+    Provides the conversion to `bytes`.
 
   * `fn __unicode__(&self) -> PyResult<PyUnicode>`
 
-    On Python 2.7, provides the conversion to `unicode`.
-    On Python 3.x, `__unicode__` is allowed but has no effect.
+    `__unicode__` is allowed but has no effect.
```
[the remark this answers is not in the document]
Comment (ijl) [reply guide/src/class.md]: I've just removed `__unicode__`.

File: Cargo.toml [hunk 2]
```diff
@@ -39,9 +39,6 @@ version_check = "0.1.5"
 [features]
 default = []
 
-# Use this feature when building python2 binding.
-python2 = []
-
 # Use this feature when building python3 binding.
 python3 = []
```
[the remark this answers is not in the document]
Comment (ijl) [reply Cargo.toml]: setuptools-rust apparently assumes it. I've left it in for now with a note.

File: guide/src/module.md [hunk 3]
```diff
@@ -53,32 +53,28 @@ Which means that the above Python code will print `This module is implemented in
 In python, modules are first class objects. This means can store them as values or add them to dicts or other modules:
 
 ```rust
-# extern crate pyo3;
```
[the remark this answers is not in the document]
Comment (ijl) [reply guide/src/module.md]: I don't see the doc-test failures locally (on 3.7). I moved the test to the Linux 3.7 job, and it's failing in a different way. Hm.

[the remark this answers is not in the document]
Comment (ijl) [reply guide/src/module.md]: Resolved with the last commit.

**Comment by ijl** [comment]
@konstin please take a look.

File: Cargo.toml (hunk 2 above)
[the remark this answers is not in the document]
Comment (konstin) [reply Cargo.toml]: Oh right. I've created https://github.com/PyO3/setuptools-rust/pull/53 for fixing that long term, but for now a comment is good

**Comment by konstin** [comment]
Please run `cargo fmt`, that should make ci pass

**Comment by ijl** [comment]
@konstin ok, the build's green.

**Comment by konstin** [comment]
Thank you!

**Pull request merged**